helpers.go

```diff
@@ -267,29 +268,111 @@ func getOffer(header string, isAccepted func(spec, offer string) bool, offers ..
 		return offers[0]
 	}
 
-	for _, offer := range offers {
-		if len(offer) == 0 {
-			continue
+	type acceptedType struct {
+		spec        string
+		quality     float64
+		specificity int
+		order       int
+	}
+
+	// Parse header and get accepted types with their quality and specificity
+	// See: https://www.rfc-editor.org/rfc/rfc9110#name-content-negotiation-fields
+	spec, commaPos, order := "", 0, 0
+	acceptedTypes := make([]acceptedType, 0, 20)
+	for len(header) > 0 {
+		order++
+
+		// Skip spaces
+		header = utils.TrimLeft(header, ' ')
+
+		// Get spec
+		commaPos = strings.IndexByte(header, ',')
+		if commaPos != -1 {
+			spec = utils.Trim(header[:commaPos], ' ')
+		} else {
+			spec = utils.TrimLeft(header, ' ')
+		}
+
+		// Get quality
+		quality := 1.0
+		if factorSign := strings.IndexByte(spec, ';'); factorSign != -1 {
+			factor := utils.Trim(spec[factorSign+1:], ' ')
+			if strings.HasPrefix(factor, "q=") {
+				if q, err := strconv.ParseFloat(factor[2:], 64); err == nil {
```

Comment:
can you test this

https://github.com/valyala/fasthttp/blob/f0865d4aabbbea51a81d56ab31a3de2dfc5a9b05/bytesconv.go#LL201C15-L201C15

maybe its faster

Reply:
seems to be

Reply:
Can you try https://github.com/gofiber/fiber/blob/master/utils/convert_s2b_new.go#L11

UnsafeBytes method instead of the other convert before you use the fasthttp method

Reply:
UnsafeBytes works great!